src/components/Approval/Approval.tsx

```diff
@@ -157,6 +146,38 @@ export const Approval = () => {
     showErrorToast,
   ])
 
+  const toggleIsExactAllowance = useCallback(() => {
+    setIsExactAllowance(!isExactAllowance)
+  }, [isExactAllowance])
+
+  useEffect(() => {
+    setBuildApprovalTxInput(undefined)
+    createBuildApprovalTxInput(isExactAllowance)
+      .then(buildApprovalTxInput => {
+        setBuildApprovalTxInput(buildApprovalTxInput)
+      })
+      .catch(e => {
+        showErrorToast(e)
+        return history.push(TradeRoutePaths.Input)
+      })
+  }, [isExactAllowance, createBuildApprovalTxInput, history, showErrorToast])
+
+  const approvalNetworkFeeCryptoHuman = useMemo(() => {
+    if (!buildApprovalTxInput) return
+
+    const { gasLimit, gasPrice, maxFeePerGas } = buildApprovalTxInput
+
+    // use EIP-1551 gas where possible
+    const totalGasPrice = maxFeePerGas ?? gasPrice
+
+    if (!totalGasPrice) return
```

Comment:
Should be able to safely use the non-null assertion operator instead, one of these two will always be defined

https://github.com/shapeshift/web/blob/4d6904e8d529ee68e21bb85c4d961b1566c0542c/packages/chain-adapters/src/evm/EvmBaseAdapter.ts#L240-L242